channel/handlers.go

```diff
@@ -3405,20 +3403,17 @@ func (server *Server) handlePlayerConnectedNotifications(conn mnet.Server, reade
 		}
 	}
 
-	for i, v := range server.players {
-		if v.ID == playerID {
-			continue
-		} else if v.hasBuddy(playerID) {
+	server.players.observe(func(plr *Player) {
```

Comment:
This appears to iterate over _all_ of the players in the channel to identify buddies and sends the packets to the targets. Would it not be better to simply iterate over the players buddy list and if any are online send it to them? A buddy list is generally ~30 characters while the active player count in a channel could be much larger. I suppose ideally, the world server would handle this logic by looking at the players buddies, identify which are online & what channel they are on, and send the packets specifically to the appropriate channels.

Reply:
So this is the handler for when the world server reports to channel servers that a player of id with name, guild id etc has loged in. We therefore have to iterate over all players as only 1 channel will have the player.

EDIT: I think this has highlited that the handlers for client and server should be split into different files for the channel server as it's gotten too long

Reply:
Ah ok. I've wondered if using a map for channels might be better, because then we could assign the channel to what it's supposed to be and target channels specifically. Right now channels are all in a slice, which is fine, but it makes it so it's harder to target a channel. If they were in a map[int]*Channel we could then just look at what channel # the player's buddies(via DB) are in and send the login packets to them specifically. 

Currently, the world server tells the channel what #(id) it should be:
https://github.com/Hucaru/Valhalla/blob/master/world/handlers.go#L82

That could be changed to where when a channel registers it tells the world what channel it wants to register as, and if 0/null then fallback to 1...20 .